shadowsocks/udp.go

```diff
@@ -390,6 +420,9 @@ func timedCopy(clientAddr net.Addr, clientConn net.PacketConn, cipher shadowaead
 			logger.Debugf("UDP Error: %v: %v", connError.Message, connError.Cause)
 			status = connError.Status
 		}
+		if expired {
```

Comment:
Why does this check have to move here?

Reply:
Moving the check prevents a spurious call to `AddUDPPacketFromTarget` on shutdown.  This isn't important for the manager metrics, but it was interfering with the tests that check how many packets have been delivered.